lib/search/document.js

```diff
@@ -0,0 +1,213 @@
+/*!
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module search/document
+ */
+
+'use strict';
+
+/**
+ * @type {module:search/field}
+ * @private
+ */
+var Field = require('./field.js');
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+/**
+ * Create a Document object to create or manipulate a document from your index.
+ *
+ * @constructor
+ * @alias module:search/document
+ *
+ * @param {string=} id - ID of the document.
+ *
+ * @example
+ * var gcloud = require('gcloud');
+ *
+ * var search = gcloud.search({
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var document = search.index('records').document('stephen');
+ */
+function Document(index, id) {
+  this.search_ = index.search_;
+  this.index_ = index;
+
+  this.id = id;
+  this.fields = {};
+}
+
+/**
+ * Add a field to this document.
+ *
+ * @throws {error} if a name is not provided.
+ *
+ * @param {string} name - This field's name.
+ * @return {module:search/field}
+ *
+ * @example
+ * var scoreField = document.addField('score');
+ * // scoreField is a Field object.
+ */
+Document.prototype.addField = function(name) {
```

Comment:
So the idea here is that you add the fields to the document, and then call Index#createDocument with the document object and the fields will be included? Yes? That's unclear here, so perhaps is there some way to improve the documentation in that regard?